plugins/modules/foreman_role.py

```diff
@@ -101,7 +118,20 @@ def main():
         if 'organizations' in entity_dict:
             entity_dict['organizations'] = module.find_resources_by_name('organizations', entity_dict['organizations'], thin=True)
 
-    changed = module.ensure_entity_state('roles', entity_dict, entity)
+    filters = entity_dict.pop("filters", None)
+
+    changed, entity = module.ensure_entity('roles', entity_dict, entity)
+
+    if not module.desired_absent and not module.check_mode:
+        if filters is not None:
+            for role_filter in filters:
+                role_filter['role_id'] = entity['id']
+                role_filter['permissions'] = module.find_resources_by_name('permissions', role_filter['permissions'])
+                module.ensure_entity_state('filters', role_filter, None, None, 'present', filters_entity_spec)
+            for old_filter in entity['filter_ids']:
+                module.ensure_entity('filters', None, {'id': old_filter}, {}, 'absent')
+            if len(entity['filter_ids']) != len(filters):
```

Comment:
this will not work if you remove and add exactly the same number of (different) filters.

what is the exact reason you don't update existing filters?

Reply:
That would require pulling each existing filter and compare them field by field with what is in YAML. That would be a much more complex implementation that would indeed result in better update tracking, and I simply can’t invest much time into doing it right now. My suggestion is to tackle it as a next iteration in a follow up pull request.

Reply:
that's fine for me, yeah. just wanted to be sure the downside is known.